.eslintrc.js

```diff
@@ -1,51 +1,42 @@
 module.exports = {
-	"env": {
-		"commonjs": true,
-		"es6": true,
-		"node": true
-	},
-	"extends": "eslint:recommended",
-	"overrides": [
-		{
-			"files": ["**/*.ts"],
-			"parser": "@typescript-eslint/parser",
-			"extends": [
-				"eslint:recommended",
-				"plugin:@typescript-eslint/eslint-recommended",
-				"plugin:@typescript-eslint/recommended"
-			],
-			"plugins": ["@typescript-eslint"],
-			"rules": {
-				"@typescript-eslint/camelcase": "off",
-				"@typescript-eslint/no-this-alias": "off",
-				"@typescript-eslint/no-var-requires": "off",
-				"@typescript-eslint/no-explicit-any": "off"
-			}
-		}
-	],
-	"globals": {
-		"Atomics": "readonly",
-		"SharedArrayBuffer": "readonly"
-	},
-	"parserOptions": {
-		"ecmaVersion": 2018
-	},
-	"rules": {
-		"indent": [
-			"error",
-			"tab"
-		],
-		"linebreak-style": [
-			"error",
-			"unix"
-		],
-		"quotes": [
-			"error",
-			"double"
-		],
-		"semi": [
-			"error",
-			"always"
-		]
-	}
+  parser: "@typescript-eslint/parser",
+  parserOptions: {
+    sourceType: "module",
+  },
+  env: {
+    commonjs: true,
+    es6: true,
+    node: true,
+  },
+  plugins: ["@typescript-eslint/eslint-plugin"],
+  extends: [
+    "eslint:recommended",
+    "plugin:@typescript-eslint/eslint-recommended",
+    "plugin:@typescript-eslint/recommended",
+  ],
+  rules: {
+    "@typescript-eslint/interface-name-prefix": "off",
+    "@typescript-eslint/explicit-function-return-type": "off",
+    "@typescript-eslint/no-explicit-any": "off",
+    "@typescript-eslint/no-unused-vars": "off",
+    "@typescript-eslint/camelcase": "off",
+    "@typescript-eslint/no-this-alias": "off",
+    "@typescript-eslint/no-var-requires": "off",
+    "@typescript-eslint/no-explicit-any": "off",
+    "@typescript-eslint/comma-spacing": "error",
+    "object-curly-spacing": ["error", "always"],
+    "space-before-function-paren": ["error", "always"],
+    "space-before-blocks": ["error", "always"],
+    "no-multi-spaces": "error",
+    "space-in-parens": "error",
+    indent: ["error", "tab"],
+    "linebreak-style": ["error", "unix"],
+    quotes: ["error", "double"],
+	semi: ["error", "always"],
+	"quote-props": ["error", "always"],
```

Comment:
Formatting got messed up here. This is also something that ESLint should catch and throw an error about. So before fixing it. Please make sure that the linter currently throws an error here.

Reply:
Added .eslintrc.js to be explicitly included since by default it's ignored by eslint.

Reply:
`interface-name-prefix` yeah I guess it's not needed in this project since interface names are following typescript recommendations.

`explicit-function-return-type` by default it's just a warning. I would let it go through this PR turned off and then maybe create another one for turning it into error if you really think it will help you. Otherwise, in my opinion, it's too much pain for a (mostly none) little gain.